scripts/ansible/roles/linux/jenkins/tasks/slave-setup.yml

```diff
@@ -0,0 +1,99 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+---
+- name: Jenkins | Set default host variables (if not already set by the user)
+  block:
+    - set_fact:
+        jenkins_agent_name: "{{ ansible_hostname }}"
+      when: jenkins_agent_name is not defined
+
+    - set_fact:
+        jenkins_agent_executors_count: 1
+      when: jenkins_agent_executors_count is not defined
+
+    - set_fact:
+        jenkins_agent_label: ""
+      when: jenkins_agent_label is not defined
+
+    - set_fact:
+        jenkins_agent_root_dir: "/home/jenkins"
+      when: jenkins_agent_root_dir is not defined
+
+- name: Jenkins | Check if the node already exists
+  jenkins_script:
+    script: jenkins.model.Jenkins.getInstance().getComputer("{{ jenkins_agent_name }}")
+    url: "{{ jenkins_url }}"
+    user: "{{ jenkins_admin_name }}"
+    password: "{{ jenkins_admin_password }}"
+  register: script_output
+
+- name: Jenkins | Create the node if it doesn't exist already
+  jenkins_script:
+    script: |
+      import jenkins.model.*
+      import hudson.model.*
+      import hudson.slaves.*
+
+      Slave agent = new DumbSlave(
+          "{{ jenkins_agent_name }}",
+          "{{ jenkins_agent_root_dir }}",
+          new JNLPLauncher()
+      )
+      agent.setNumExecutors({{ jenkins_agent_executors_count }})
+      agent.setLabelString("{{ jenkins_agent_label }}")
+      agent.setMode(Node.Mode.NORMAL)
+      agent.setRetentionStrategy(new RetentionStrategy.Always())
+
+      Jenkins.instance.addNode(agent)
+
+    url: "{{ jenkins_url }}"
+    user: "{{ jenkins_admin_name }}"
+    password: "{{ jenkins_admin_password }}"
+  when: (script_output.output.lstrip('Result:') | trim) == ""
+
+- name: Jenkins | Extract the Jenkins agent secret
+  jenkins_script:
+    script: jenkins.model.Jenkins.getInstance().getComputer("{{ jenkins_agent_name }}").getJnlpMac()
+    url: "{{ jenkins_url }}"
+    user: "{{ jenkins_admin_name }}"
+    password: "{{ jenkins_admin_password }}"
+  register: script_output
+
+- set_fact:
+    node_secret: "{{ script_output.output.lstrip('Result:') | trim }}"
+
+- name: Jenkins | Install Java JRE needed by Jenkins
+  apt:
+    name: openjdk-8-jre
+    state: present
+    update_cache: yes
+
+- name: Jenkins | Create the Jenkins group
+  group:
+    name: jenkins
+    state: present
+
+- name: Jenkins | Add the Jenkins user
+  user:
+    name: jenkins
+    group: jenkins
+    state: present
+
+- import_role:
+    name: linux/docker
+    tasks_from: stable-install.yml
```

Comment:
Yes, it does.

The `tasks_from` instance represents the `yml` file to be executed when importing the role. If this is not given, it looks for `main.yml` in the `tasks` directory for that specific role.

Currently, we have the following roles directory layout:
```
roles
└── linux
    ├── az-dcap-client
    │   └── tasks
    │       └── stable-install.yml
    ├── common
    │   └── tasks
    │       ├── apt-repo-intel.yml
    │       ├── apt-repo-llvm.yml
    │       └── apt-repo-microsoft.yml
    ├── docker
    │   └── tasks
    │       ├── ci-setup.yml
    │       └── stable-install.yml
    ├── intel
    │   └── tasks
    │       ├── sgx-driver.yml
    │       └── sgx-packages.yml
    ├── jenkins
    │   └── tasks
    │       └── slave-setup.yml
    └── openenclave
        └── tasks
            ├── environment-setup.yml
            └── stable-install.yml
```
I've given meaningful names for the `yml` files, and instead of relying on the expected `main.yml` file when importing the role, I always specified what exactly I wanted to be run from the role.